packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -81,7 +150,29 @@ const Indicator = () => {
       buildId !== newBuildInfo?.latestBuild?.id &&
       currentBuild?.buildStatus === BuildStatus.SUCCESS
     ) {
-      setBuildInfo({ ...newBuildInfo, buildStatus: BuildStatus.SUCCESS })
+      if (refreshNeeded) {
+        setBuildInfo({ ...newBuildInfo, buildStatus: BuildStatus.SUCCESS })
+      } else {
+        const { hasPageChanged, errorMessage } = await hasPageDataChanged(
+          buildId
+        )
+
+        if (errorMessage) {
```

Comment:
Ooof, these nested blocks are getting gnarly :skull: Can't wait to get some refactoring going on! (no action needed, I'm just looking forward on making ALL of the existing logic easier to read and understand in the future.)

Reply:
Can't agree more. all this needs some serious UI love